File: backend/src/main/java/com/portfolioapp/portfoliobackend/dto/PhotoDto.java
package com.portfolioapp.portfoliobackend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PhotoDto {
    private Long id;
    private String title;
    private String description;
    private String imageUrl;
    private String author; // email автора

    private String authorFullName;
    private String authorAvatarUrl;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @Builder.Default
    private int likesCount = 0;
    @Builder.Default
    private boolean likedByCurrentUser = false;
}

File: app/src/main/java/com/example/portfolioapp/network/PhotoDto.kt
package com.example.portfolioapp.network

import com.google.gson.annotations.SerializedName

data class PhotoDto(
    val id: Long,
    val title: String,
    val description: String,
    val imageUrl: String,

    @SerializedName("author") val author: String = "", // email
    @SerializedName("authorFullName") val authorFullName: String? = null, // ✅ НОВОЕ
    @SerializedName("authorAvatarUrl") val authorAvatarUrl: String? = null, // ✅ НОВОЕ

    val createdAt: String,
    val updatedAt: String,

    @SerializedName("likesCount") val likesCount: Int = 0,
    @SerializedName("likedByCurrentUser") val likedByCurrentUser: Boolean = false
)